File: examples/sockets/tcp/objectserver/MyCalculation.java
package tcp.objectserver;

public class MyCalculation extends WorkRequest
{

    private static final long serialVersionUID = -782657330809139571L;
    private int n;
    private int[] values;


    public MyCalculation(int n) {
	this.n = n;
	values = new int[n];
    }


    public Object execute() {
	for (int i = 0; i < values.length; i++)
	    values[i] = i * i;
	return values;
    }
}

File: examples/golang/sockets/tcp/objectserver/MyCalculation.go
package main
import "fmt"
type MyCalculation struct {
	Size int
	Values []int
}

func NewMyCalculation(n int) *MyCalculation {
	mc := MyCalculation{Size: n}
	return &mc
}

func (m MyCalculation) Execute() any {
	m.Values = make([]int, m.Size)
	fmt.Println("MyCalculation: ", m.Size)
	for i, _ := range m.Values {
		m.Values[i] = i * i
	}
	return m.Values
}
